cardano-cli/src/Cardano/CLI/Render.hs

```diff
@@ -0,0 +1,48 @@
+module Cardano.CLI.Render
+  ( customRenderHelp
+  ) where
+
+import           Cardano.Prelude
+import           Data.Function (id)
+import           Options.Applicative
+import           Options.Applicative.Help.Ann
+import           Options.Applicative.Help.Types (helpText)
+import           Prelude (String)
+import           Prettyprinter
+import           Prettyprinter.Render.Util.SimpleDocTree
+
+import qualified Data.Text as T
+import qualified System.Environment as IO
+import qualified System.IO.Unsafe as IO
+
+cliHelpTraceEnabled :: Bool
+cliHelpTraceEnabled = IO.unsafePerformIO $ do
```

Comment:
I don't have any experience using `unsafePerformIO`. Is this ok to use here?

Reply:
Yes this seems to be a safe use of `unsafePerformIO` .

Reply:
Yep.  This example of `unsafePerformIO` is fairly normal.  `unsafePerformIO` is used to perform some IO to initialise a global constant.  The danger is if the side effect is executed more than once, in which case the constant is not so constant, but this is prevent by the `NOINLINE` pragma.  This forces the value to be a thunk that can only be evaluated once.

Reply:
See documentation on `unsafePerformIO`: https://hackage.haskell.org/package/base-4.15.0.0/docs/System-IO-Unsafe.html